btm/src/main/java/bitronix/tm/resource/common/XAPool.java

```diff
@@ -285,12 +304,13 @@ private synchronized void createPooledObject(Object xaFactory) throws Exception
         objects.add(xaStatefulHolder);
     }
 
-    private static Object createXAFactory(ResourceBean bean) throws Exception {
+    private static Object createXAFactory(ResourceBean bean, Object[] factoryConstructorArgs) throws Exception {
         String className = bean.getClassName();
         if (className == null)
             throw new IllegalArgumentException("className cannot be null");
         Class xaFactoryClass = ClassLoaderUtils.loadClass(className);
-        Object xaFactory = xaFactoryClass.newInstance();
+
+        Object xaFactory = invokeConstructor(xaFactoryClass, factoryConstructorArgs);
```

Comment:
i think it's better to avoid the commons-lang dependency and do this the old ugly way